services/order.py

```diff
@@ -0,0 +1,35 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict],
+    username: str,
+    date: str = None
+) -> Order:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+        order.save()
```

Comment:
Don't Repeat Yourself: In the `create_order` function, you're calling the `save()` method twice if the `date` parameter is not `None`. This is a bit inefficient. It would be better to set all the attributes you want and then call `save()` only once. For example, you could move the `order.save()` call to the end of the function, just before the `return` statement.